components/scream/src/physics/shoc/shoc_functions.hpp

```diff
@@ -86,6 +86,11 @@ struct Functions
     Scalar& thl_sec, Scalar& qw_sec, Scalar& wthl_sec, Scalar& wqw_sec,
     Scalar& qwthl_sec, Scalar& uw_sec, Scalar& vw_sec, Scalar& wtke_sec);
 
+  KOKKOS_FUNCTION
+  static void shoc_pblintd_cldcheck(
```

Comment:
Right now, this function consists entirely of an if test applied to `cldn`, and a related assignment. We could pass it `uview_1d` arguments for the entire column and pull the loop inside this function, which would allow us to use a mask instead.

Reply:
ok, I will do it in that way though

Reply:
@jeff-cohere @ambrad In this way, we are going to endup repack the data to the same packsize in the inner loop, I was wondering whether this is efficient way to do, because the pblh is 1-dimensional data, but zi, and cldn are 2-dimensional.

Reply:
@jeff-cohere see my comment above. Does this resolve your initial comment?

Reply:
Right, yes. It only happens once per column. Sorry for the confusion!